### PR TITLE
Add status() docs to Client JS reference

### PR DESCRIPTION
**Pull request by johndmulhausen**
What it says on the tin

### CODE REVIEW

**Comment by filecorgi** [comment]
- Image optimization came back clean!
- Vuepress build was successful!
### `docs/how-tos/regenerate-api-key.md`:
- The following link failed to connect:
  https://status.web3.storage
- The following link returned HTTP status code 404 (Not Found):
  https://staging.web3.storage/legal

### `docs/how-tos/retrieve.md`:
- The following link failed to connect:
  https://status.web3.storage
- The following link returned HTTP status code 404 (Not Found):
  https://staging.web3.storage/legal
- If a new sentence starts here, add a space and start with an uppercase letter.
  ```
  ... you have a client, you can call client.get, passing in a CID string:  The g...
                                             ^^^
  ```
- An apostrophe may be missing.
  ```
  ...access to the retrieved IPFS data.  The files method returns an array of Web...
                                             ^^^^^
  ```
- If a new sentence starts here, add a space and start with an uppercase letter.
  ```
  ...is guide, we'll use the gateway at dweb.link, but you can check the list of ...
                                             ^^^^
  ```
### `docs/how-tos/store.md`:
- The following link failed to connect:
  https://web3.storage/tokens/
- The following link failed to connect:
  https://web3.storage/files/
- The following link failed to connect:
  https://status.web3.storage
- The following link returned HTTP status code 404 (Not Found):
  https://staging.web3.storage/legal
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ...APIs like fetch  (opens new window) and File  (opens new window). This guide...
                                             ^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ...ile objects manually: Unique file names Try to give each file a unique file ...
                                             ^^^
  ```
- Apostrophe might be missing. Did you mean “file’s” (curly possessive apostrophe) or “file's” (straight possessive apostrophe)? (Some terms such as “farmers’ market” have a possessive apostrophe after the first noun.)
  ```
  ...one content archive, and linking to the files inside is much simpler if each...
                                             ^^^^^
  ```
- Consider using a more formal alternative to avoid the overused word “more”.
  ```
  ...the IPFS command line. You can also get more information about the status of...
                                             ^^^^^^^^^^^^^^^^
  ```

### `docs/reference/client-library.md`:
- The following link failed to connect:
  https://status.web3.storage
- The following link returned HTTP status code 404 (Not Found):
  https://staging.web3.storage/legal
- Possibly missing preposition found.
  ```
  ... the JavaScript client library for Web3.Storage, you must first obtain an AP...
                                             ^^^^^^^
  ```
- If a new sentence starts here, add a space and start with an uppercase letter.
  ```
  ... the iterator represent files. If entry.type == 'directory', the entry repre...
                                             ^^^^
  ```
- Possibly missing comma found.
  ```
  ...tional order. Number Type Description 1 string A string containing the CID o...
                                             ^^^^^^
  ```
- Possibly missing comma found.
  ```
  ...etwork is storing your data. Using this information you can enumerate peers ...
                                             ^^^^^^^^^^^
  ```
- Possibly missing comma found.
  ```
  ...tional order. Number Type Description 1 string A string containing the CID o...
                                             ^^^^^^
  ```

File: docs/reference/client-library.md [hunk 1]
```diff
@@ -156,16 +156,52 @@ Parameters are supplied in positional order.
 | ------ | ---- | ----------- |
 | 1 | `string` | A string containing the CID of the CAR to be retrieved. |
 
-## Retrieve metadata
+## Check status
 
-Retrieve metadata about your file by using the `status()` method, supplying the CID of the file you are interested in.
+Retrieve metadata about your file by using the `status()` method, supplying the CID of the file you are interested in. This metadata includes the creation date and size, as well as details about how the network is storing your data. Using this information you can enumerate peers on the InterPlanetary File System (IPFS) network that are pinning the data, and Filecoin miners that have accepted deals to store the data.
```
Comment (alanshaw) [inline docs/reference/client-library.md:161]:
```suggestion
Retrieve metadata about your file by using the `status()` method, supplying the CID of the file you are interested in. This metadata includes the creation date and size, as well as details about how the network is storing your data. Using this information you can enumerate peers on the InterPlanetary File System (IPFS) network that are pinning the data, and Filecoin storage providers that have accepted deals to store the data.
```

File: docs/reference/client-library.md [hunk 2]
```diff
@@ -50,13 +50,13 @@ Method parameters are supplied in positional order.
 | Number | Type | Description |
 | ------ | ---- | ----------- |
 | 1 | `file[]` | An iterable collection of [Files](https://developer.mozilla.org/en-US/docs/Web/API/File) to be packed into a CAR and uploaded. |
-| 2 | `{options}` | Optional. A JSON object whose properties define certain Web3.storage options and metadata about the files being uploaded. See below for more details. |
+| 2 | `{options}` | Optional. A JSON object whose properties define certain Web3.Storage options and metadata about the files being uploaded. See below for more details. |
```
Comment (alanshaw) [inline docs/reference/client-library.md:53]: Not JSON, just a plain object.

```suggestion
| 2 | `{options}` | Optional. An object whose properties define certain Web3.Storage options and metadata about the files being uploaded. See below for more details. |
```

File: docs/reference/client-library.md [hunk 3]
```diff
@@ -50,13 +50,13 @@ Method parameters are supplied in positional order.
...
 An `{options}` object is a JSON object that defines option parameters:
 
-```js
+```json
 {
-  name: 'cat pics',
+  name: "cat pics",
   maxRetries: 3,
   onRootCidReady: handlerFunction,
   onStoredChunk: handlerFunction
```
Comment (alanshaw) [inline docs/reference/client-library.md:62]: You can find good descriptions of these options here: https://github.com/web3-storage/web3.storage/blob/20d02fc22725de825cc2a4f773b40e0d1f8a6a68/packages/client/src/lib/interface.ts#L62

**Review by johnnymatthews — COMMENTED** [review]
See comment https://github.com/web3-storage/docs/pull/30/files#r673995355

**Comment by filecorgi** [comment]
- Image optimization came back clean!
- Vuepress build was successful!
### `docs/how-tos/regenerate-api-key.md`:
- The following link failed to connect:
  https://status.web3.storage
- The following link returned HTTP status code 404 (Not Found):
  https://staging.web3.storage/legal

### `docs/how-tos/retrieve.md`:
- The following link failed to connect:
  https://status.web3.storage
- The following link returned HTTP status code 404 (Not Found):
  https://staging.web3.storage/legal
- If a new sentence starts here, add a space and start with an uppercase letter.
  ```
  ... you have a client, you can call client.get, passing in a CID string:  The g...
                                             ^^^
  ```
- An apostrophe may be missing.
  ```
  ...access to the retrieved IPFS data.  The files method returns an array of Web...
                                             ^^^^^
  ```
- If a new sentence starts here, add a space and start with an uppercase letter.
  ```
  ...is guide, we'll use the gateway at dweb.link, but you can check the list of ...
                                             ^^^^
  ```
### `docs/how-tos/store.md`:
- The following link failed to connect:
  https://web3.storage/tokens/
- The following link failed to connect:
  https://web3.storage/files/
- The following link failed to connect:
  https://status.web3.storage
- The following link returned HTTP status code 404 (Not Found):
  https://staging.web3.storage/legal
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ...APIs like fetch  (opens new window) and File  (opens new window). This guide...
                                             ^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ...ile objects manually: Unique file names Try to give each file a unique file ...
                                             ^^^
  ```
- Apostrophe might be missing. Did you mean “file’s” (curly possessive apostrophe) or “file's” (straight possessive apostrophe)? (Some terms such as “farmers’ market” have a possessive apostrophe after the first noun.)
  ```
  ...one content archive, and linking to the files inside is much simpler if each...
                                             ^^^^^
  ```
- Consider using a more formal alternative to avoid the overused word “more”.
  ```
  ...the IPFS command line. You can also get more information about the status of...
                                             ^^^^^^^^^^^^^^^^
  ```

### `docs/reference/client-library.md`:
- The following link failed to connect:
  https://status.web3.storage
- The following link returned HTTP status code 404 (Not Found):
  https://staging.web3.storage/legal
- Possibly missing preposition found.
  ```
  ... the JavaScript client library for Web3.Storage, you must first obtain an AP...
                                             ^^^^^^^
  ```
- If a new sentence starts here, add a space and start with an uppercase letter.
  ```
  ... the iterator represent files. If entry.type == 'directory', the entry repre...
                                             ^^^^
  ```
- Possibly missing comma found.
  ```
  ...tional order. Number Type Description 1 string A string containing the CID o...
                                             ^^^^^^
  ```
- Possibly missing comma found.
  ```
  ...etwork is storing your data. Using this information you can enumerate peers ...
                                             ^^^^^^^^^^^
  ```
- Possibly missing comma found.
  ```
  ...tional order. Number Type Description 1 string A string containing the CID o...
                                             ^^^^^^
  ```

**Comment by filecorgi** [comment]
- Image optimization came back clean!
- Vuepress build was successful!
### `docs/how-tos/regenerate-api-key.md`:
- The following link failed to connect:
  https://status.web3.storage
- The following link returned HTTP status code 404 (Not Found):
  https://staging.web3.storage/legal

### `docs/how-tos/retrieve.md`:
- The following link failed to connect:
  https://status.web3.storage
- The following link returned HTTP status code 404 (Not Found):
  https://staging.web3.storage/legal
- If a new sentence starts here, add a space and start with an uppercase letter.
  ```
  ... you have a client, you can call client.get, passing in a CID string:  The g...
                                             ^^^
  ```
- An apostrophe may be missing.
  ```
  ...access to the retrieved IPFS data.  The files method returns an array of Web...
                                             ^^^^^
  ```
- If a new sentence starts here, add a space and start with an uppercase letter.
  ```
  ...is guide, we'll use the gateway at dweb.link, but you can check the list of ...
                                             ^^^^
  ```
### `docs/how-tos/store.md`:
- The following link failed to connect:
  https://web3.storage/tokens/
- The following link failed to connect:
  https://web3.storage/files/
- The following link failed to connect:
  https://status.web3.storage
- The following link returned HTTP status code 404 (Not Found):
  https://staging.web3.storage/legal
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ...APIs like fetch  (opens new window) and File  (opens new window). This guide...
                                             ^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ...ile objects manually: Unique file names Try to give each file a unique file ...
                                             ^^^
  ```
- Apostrophe might be missing. Did you mean “file’s” (curly possessive apostrophe) or “file's” (straight possessive apostrophe)? (Some terms such as “farmers’ market” have a possessive apostrophe after the first noun.)
  ```
  ...one content archive, and linking to the files inside is much simpler if each...
                                             ^^^^^
  ```
- Consider using a more formal alternative to avoid the overused word “more”.
  ```
  ...the IPFS command line. You can also get more information about the status of...
                                             ^^^^^^^^^^^^^^^^
  ```

### `docs/reference/client-library.md`:
- The following link failed to connect:
  https://status.web3.storage
- The following link returned HTTP status code 404 (Not Found):
  https://staging.web3.storage/legal
- Possibly missing preposition found.
  ```
  ... the JavaScript client library for Web3.Storage, you must first obtain an AP...
                                             ^^^^^^^
  ```
- If a new sentence starts here, add a space and start with an uppercase letter.
  ```
  ... the iterator represent files. If entry.type == 'directory', the entry repre...
                                             ^^^^
  ```
- Possibly missing comma found.
  ```
  ...tional order. Number Type Description 1 string A string containing the CID o...
                                             ^^^^^^
  ```
- Possibly missing comma found.
  ```
  ...etwork is storing your data. Using this information you can enumerate peers ...
                                             ^^^^^^^^^^^
  ```
- Possibly missing comma found.
  ```
  ...tional order. Number Type Description 1 string A string containing the CID o...
                                             ^^^^^^
  ```

**Comment by yusefnapora** [comment]
@johndmulhausen / @johnnymatthews  How do you guys feel about using a table for the `Pin` and `Deal` object fields?

Screen Shot 2021-07-21 at 12 02 48 PM

I think it's a bit easier to scan through, but maybe that's just me :)

File: docs/reference/client-library.md [hunk 4]
```diff
@@ -176,12 +212,48 @@ Parameters are supplied in positional order.
 
 ### Return value
 
-The `status()` method returns a `Metadata` object that contains the metadata for your object's storage deal on the web3.storage network.
+The `status()` method returns a `{Status}` object that contains the metadata for your object's storage deal on the Web3.Storage network. This object will have the following properties:
```
Comment (yusefnapora) [inline docs/reference/client-library.md:215]: I think we should mention that it will return `undefined` if the CID isn't found.

**Review by yusefnapora — CHANGES_REQUESTED** [review]

File: docs/reference/client-library.md (hunk 4 above)
Comment (olizilla) [reply docs/reference/client-library.md:215]: Aye! I'm not sure of the ergonomics of that decision tho. We could just have it throw on anything other than a Cid that we were able to get a status for. What do you think?

Comment (johndmulhausen) [reply docs/reference/client-library.md:215]: @olizilla I feel like returning an object with really obvious "not found" values might be more merciful given that `undefined` will almost surely throw people's code into an exception state, but hey.

File: docs/reference/client-library.md (hunk 3 above)
Comment (johndmulhausen) [reply docs/reference/client-library.md:62]: Just a bit further down the parameters are covered in much better detail so I removed this bit

**Comment by filecorgi** [comment]
- Image optimization came back clean!
- Vuepress build was successful!
### `docs/reference/client-library.md`:
- The following link failed to connect:
  https://status.web3.storage
- The following link returned HTTP status code 404 (Not Found):
  https://staging.web3.storage/legal
- Possibly missing preposition found.
  ```
  ... the JavaScript client library for Web3.Storage, you must first obtain an AP...
                                             ^^^^^^^
  ```
- If a new sentence starts here, add a space and start with an uppercase letter.
  ```
  ... the iterator represent files. If entry.type == 'directory', the entry repre...
                                             ^^^^
  ```
- Possibly missing comma found.
  ```
  ...tional order. Number Type Description 1 string A string containing the CID o...
                                             ^^^^^^
  ```
- Possibly missing comma found.
  ```
  ...etwork is storing your data. Using this information you can enumerate peers ...
                                             ^^^^^^^^^^^
  ```
- Possibly missing comma found.
  ```
  ...tional order. Number Type Description 1 string A string containing the CID o...
                                             ^^^^^^
  ```

**Comment by filecorgi** [comment]
- Image optimization came back clean!
- Vuepress build was successful!
### `docs/reference/client-library.md`:
- The following link failed to connect:
  https://status.web3.storage
- The following link returned HTTP status code 404 (Not Found):
  https://staging.web3.storage/legal
- Possibly missing preposition found.
  ```
  ... the JavaScript client library for Web3.Storage, you must first obtain an AP...
                                             ^^^^^^^
  ```
- ‘your’ (possessive pronoun) seems less likely than ‘you’ (personal pronoun).
  ```
  ...ties that can be used as parameters for your put() operation:  String. The n...
                                             ^^^^
  ```
- If a new sentence starts here, add a space and start with an uppercase letter.
  ```
  ... the iterator represent files. If entry.type == 'directory', the entry repre...
                                             ^^^^
  ```
- Possibly missing comma found.
  ```
  ...tional order. Number Type Description 1 string A string containing the CID o...
                                             ^^^^^^
  ```
- Possibly missing comma found.
  ```
  ...etwork is storing your data. Using this information you can enumerate peers ...
                                             ^^^^^^^^^^^
  ```
- Possibly missing comma found.
  ```
  ...tional order. Number Type Description 1 string A string containing the CID o...
                                             ^^^^^^
  ```

**Comment by filecorgi** [comment]
- Image optimization came back clean!
- Vuepress build was successful!
### `docs/reference/client-library.md`:
- The following link failed to connect:
  https://status.web3.storage
- The following link returned HTTP status code 404 (Not Found):
  https://staging.web3.storage/legal
- Possibly missing preposition found.
  ```
  ... the JavaScript client library for Web3.Storage, you must first obtain an AP...
                                             ^^^^^^^
  ```
- ‘your’ (possessive pronoun) seems less likely than ‘you’ (personal pronoun).
  ```
  ...ties that can be used as parameters for your put() operation:  String. The n...
                                             ^^^^
  ```
- If a new sentence starts here, add a space and start with an uppercase letter.
  ```
  ... the iterator represent files. If entry.type == 'directory', the entry repre...
                                             ^^^^
  ```
- Possibly missing comma found.
  ```
  ...tional order. Number Type Description 1 string A string containing the CID o...
                                             ^^^^^^
  ```
- Possibly missing comma found.
  ```
  ...etwork is storing your data. Using this information you can enumerate peers ...
                                             ^^^^^^^^^^^
  ```
- Possibly missing comma found.
  ```
  ...tional order. Number Type Description 1 string A string containing the CID o...
                                             ^^^^^^
  ```

**Comment by johndmulhausen** [comment]
> @johndmulhausen / @johnnymatthews How do you guys feel about using a table for the `Pin` and `Deal` object fields?
> 
> Screen Shot 2021-07-21 at 12 02 48 PM
> 
> I think it's a bit easier to scan through, but maybe that's just me :)

Just confirming that @johnnymatthews , @yusefnapora , and I all had a great chat about this issue and decided to leave this in object form as it just scans *on the page* really well and does a better job visually representing the structure of `Deal` and `Pin` objects.

**Comment by filecorgi** [comment]
- Image optimization came back clean!
- Vuepress build was successful!
### `docs/reference/client-library.md`:
- The following link failed to connect:
  https://status.web3.storage
- The following link returned HTTP status code 404 (Not Found):
  https://staging.web3.storage/legal
- Possibly missing preposition found.
  ```
  ... the JavaScript client library for Web3.Storage, you must first obtain an AP...
                                             ^^^^^^^
  ```
- ‘your’ (possessive pronoun) seems less likely than ‘you’ (personal pronoun).
  ```
  ...ties that can be used as parameters for your put() operation:  String. The n...
                                             ^^^^
  ```
- If a new sentence starts here, add a space and start with an uppercase letter.
  ```
  ... the iterator represent files. If entry.type == 'directory', the entry repre...
                                             ^^^^
  ```
- Possibly missing comma found.
  ```
  ...tional order. Number Type Description 1 string A string containing the CID o...
                                             ^^^^^^
  ```
- Possibly missing comma found.
  ```
  ...tional order. Number Type Description 1 string A string containing the CID o...
                                             ^^^^^^
  ```

**Review by johnnymatthews — APPROVED** [review]

File: docs/reference/client-library.md (hunk 4 above)
Comment (yusefnapora) [reply docs/reference/client-library.md]: I like the idea of a "not found" response object a bit better than throwing, but either way could work. For now it seems worth documenting the current behavior & we can update if the API changes.

Comment (olizilla) [reply docs/reference/client-library.md]: For context, I took my cues from the behaviour of `Map.get()` here. It returns `undefined` if you ask for a thing that is not in it. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/get

**Comment by filecorgi** [comment]
- Image optimization came back clean!
- Vuepress build was successful!
### `docs/reference/client-library.md`:
- The following link failed to connect:
  https://status.web3.storage
- The following link returned HTTP status code 404 (Not Found):
  https://staging.web3.storage/legal
- Possibly missing preposition found.
  ```
  ... the JavaScript client library for Web3.Storage, you must first obtain an AP...
                                             ^^^^^^^
  ```
- ‘your’ (possessive pronoun) seems less likely than ‘you’ (personal pronoun).
  ```
  ...ties that can be used as parameters for your put() operation:  String. The n...
                                             ^^^^
  ```
- If a new sentence starts here, add a space and start with an uppercase letter.
  ```
  ... the iterator represent files. If entry.type == 'directory', the entry repre...
                                             ^^^^
  ```
- Possibly missing comma found.
  ```
  ...tional order. Number Type Description 1 string A string containing the CID o...
                                             ^^^^^^
  ```
- Possibly missing comma found.
  ```
  ...tional order. Number Type Description 1 string A string containing the CID o...
                                             ^^^^^^
  ```

File: docs/reference/client-library.md (hunk 4 above)
Comment (johndmulhausen) [reply docs/reference/client-library.md]: Agreed on doc'ing current behavior. Added details for `undefined` for `status()` and `get()`. 

I ain't mad at the method returning `undefined` -- stark as it is. I just have a hunch that a "not found" response object with a helpful message embedded is somewhat more akin to, for instance, an HTTP status code of 500 with details in the payload. Food for thought, anyway. :)

**Review by yusefnapora — APPROVED** [review]

**Pull request merged**